provider-proxy/README.md

```diff
@@ -0,0 +1,7 @@
+# `provider-proxy`
+
+> [!NOTE]
+> 
+> ***This package is for people building TensorZero itself, not for people using TensorZero for their applications.***
+
+This is a caching MITM proxy, used to cache (notoriously flaky) model provider requests for our E2E tests.
```

Comment:
not obvious how I run the tests against this

Reply:
I added instructions to the README